tests/ui/match_overlapping_arm.rs

```diff
@@ -69,6 +69,24 @@ fn overlapping() {
         _ => (),
     }
 
+    match 42 {
+        0..14 => println!("0 .. 14"),
+        5..10 => println!("5 .. 10"),
+        _ => (),
+    }
```

Comment:
Huh, shouldn't rustc trigger here? the second arm is unreachable. Well, at least this lint doesn't trigger here. 

A quick check on [Playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=e71febb8617c05922852c5c1e153fe6f) shows that rustc triggers here correctly. So I assume this is a late pass lint in rustc and therefore would only trigger once the clippy lint doesn't trigger anymore.

Reply:
Yes I was a bit surprise too, I was also waiting for an unreachable warning :sweat_smile: